[Bug report] Trino, create table for remote IRC failed caused by: Unable to load credentials from any of the providers in the chain AwsCredentialsProviderChain

### Version

main branch

### Describe what's wrong


```
trino:mydb_s3> CREATE SCHEMA remote_irc.product_s3; 
CREATE SCHEMA
trino:mydb_s3> use remote_irc.product_s3;
USE
trino:product_s3> CREATE TABLE page_views ( view_time TIMESTAMP, user_id BIGINT, page_url VARCHAR, ds DATE, country VARCHAR)WITH ( format = 'ORC', partitioning = ARRAY['ds', 'country'], sorted_by = ARRAY['user_id']); 
Query 20260429_064726_00014_brhru failed: Failed checking new table's location: s3://irc-e2e-env1-metalake-b-warehouse/env1/metalake_b/catalog_1/product_s3/page_views-c146b228bfd646889d20aeab7db054b0
io.trino.spi.TrinoException: Failed checking new table's location: s3://irc-e2e-env1-metalake-b-warehouse/env1/metalake_b/catalog_1/product_s3/page_views-c146b228bfd646889d20aeab7db054b0
	at io.trino.plugin.iceberg.IcebergMetadata.beginCreateTable(IcebergMetadata.java:1318)
	at io.trino.plugin.iceberg.IcebergMetadata.createTable(IcebergMetadata.java:1196)
	at io.trino.plugin.base.classloader.ClassLoaderSafeConnectorMetadata.createTable(ClassLoaderSafeConnectorMetadata.java:461)
	at io.trino.tracing.TracingConnectorMetadata.createTable(TracingConnectorMetadata.java:388)
	at io.trino.metadata.MetadataManager.createTable(MetadataManager.java:878)
	at io.trino.tracing.TracingMetadata.createTable(TracingMetadata.java:436)
	at io.trino.execution.CreateTableTask.internalExecute(CreateTableTask.java:306)
	at io.trino.execution.CreateTableTask.execute(CreateTableTask.java:128)
	at io.trino.execution.CreateTableTask.execute(CreateTableTask.java:94)
	at io.trino.execution.DataDefinitionExecution.start(DataDefinitionExecution.java:152)
	at io.trino.execution.SqlQueryManager.createQuery(SqlQueryManager.java:284)
	at io.trino.dispatcher.LocalDispatchQuery.startExecution(LocalDispatchQuery.java:150)
	at io.trino.dispatcher.LocalDispatchQuery.lambda$waitForMinimumWorkers$1(LocalDispatchQuery.java:134)
	at io.airlift.concurrent.MoreFutures.lambda$addSuccessCallback$0(MoreFutures.java:570)
	at io.airlift.concurrent.MoreFutures$3.onSuccess(MoreFutures.java:545)
	at com.google.common.util.concurrent.Futures$CallbackListener.run(Futures.java:1132)
	at io.trino.$gen.Trino_478____20260428_160509_2.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
	at java.base/java.lang.Thread.run(Thread.java:1474)
Caused by: io.trino.filesystem.TrinoFileSystemException: Failed to list location: s3://irc-e2e-env1-metalake-b-warehouse/env1/metalake_b/catalog_1/product_s3/page_views-c146b228bfd646889d20aeab7db054b0
	at io.trino.filesystem.s3.S3FileSystem.listObjects(S3FileSystem.java:270)
	at io.trino.filesystem.s3.S3FileSystem.listFiles(S3FileSystem.java:240)
	at io.trino.filesystem.switching.SwitchingFileSystem.listFiles(SwitchingFileSystem.java:116)
	at io.trino.filesystem.tracing.TracingFileSystem.lambda$listFiles$0(TracingFileSystem.java:119)
	at io.trino.filesystem.tracing.Tracing.withTracing(Tracing.java:51)
	at io.trino.filesystem.tracing.TracingFileSystem.listFiles(TracingFileSystem.java:119)
	at io.trino.filesystem.cache.CacheFileSystem.listFiles(CacheFileSystem.java:106)
	at io.trino.plugin.iceberg.IcebergMetadata.beginCreateTable(IcebergMetadata.java:1309)
	... 19 more
Caused by: software.amazon.awssdk.core.exception.SdkClientException: Unable to load credentials from any of the providers in the chain AwsCredentialsProviderChain(credentialsProviders=[SystemPropertyCredentialsProvider(), EnvironmentVariableCredentialsProvider(), WebIdentityTokenCredentialsProvider(), ProfileCredentialsProvider(profileName=default, profileFile=ProfileFile(sections=[])), ContainerCredentialsProvider(), InstanceProfileCredentialsProvider()]) : [SystemPropertyCredentialsProvider(): Unable to load credentials from system settings. Access key must be specified either via environment variable (AWS_ACCESS_KEY_ID) or system property (aws.accessKeyId)., EnvironmentVariableCredentialsProvider(): Unable to load credentials from system settings. Access key must be specified either via environment variable (AWS_ACCESS_KEY_ID) or system property (aws.accessKeyId)., WebIdentityTokenCredentialsProvider(): Either the environment variable AWS_WEB_IDENTITY_TOKEN_FILE or the javaproperty aws.webIdentityTokenFile must be set., ProfileCredentialsProvider(profileName=default, profileFile=ProfileFile(sections=[])): Profile file contained no credentials for profile 'default': ProfileFile(sections=[]), ContainerCredentialsProvider(): Cannot fetch credentials from container - neither AWS_CONTAINER_CREDENTIALS_FULL_URI or AWS_CONTAINER_CREDENTIALS_RELATIVE_URI environment variables are set., InstanceProfileCredentialsProvider(): Failed to load credentials from IMDS.]
	at software.amazon.awssdk.core.exception.SdkClientException$BuilderImpl.build(SdkClientException.java:130)
	at software.amazon.awssdk.auth.credentials.AwsCredentialsProviderChain.resolveCredentials(AwsCredentialsProviderChain.java:130)
	at software.amazon.awssdk.auth.credentials.internal.LazyAwsCredentialsProvider.resolveCredentials(LazyAwsCredentialsProvider.java:45)
	at software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider.resolveCredentials(DefaultCredentialsProvider.java:134)
	at software.amazon.awssdk.auth.credentials.AwsCredentialsProvider.resolveIdentity(AwsCredentialsProvider.java:54)
	at software.amazon.awssdk.services.s3.auth.scheme.internal.S3AuthSchemeInterceptor.lambda$trySelectAuthScheme$6(S3AuthSchemeInterceptor.java:169)
	at software.amazon.awssdk.core.internal.util.MetricUtils.reportDuration(MetricUtils.java:81)
	at software.amazon.awssdk.services.s3.auth.scheme.internal.S3AuthSchemeInterceptor.trySelectAuthScheme(S3AuthSchemeInterceptor.java:169)
	at software.amazon.awssdk.services.s3.auth.scheme.internal.S3AuthSchemeInterceptor.selectAuthScheme(S3AuthSchemeInterceptor.java:87)
	at software.amazon.awssdk.services.s3.auth.scheme.internal.S3AuthSchemeInterceptor.beforeExecution(S3AuthSchemeInterceptor.java:67)
	at software.amazon.awssdk.core.interceptor.ExecutionInterceptorChain.lambda$beforeExecution$1(ExecutionInterceptorChain.java:59)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1604)
	at software.amazon.awssdk.core.interceptor.ExecutionInterceptorChain.beforeExecution(ExecutionInterceptorChain.java:59)
	at software.amazon.awssdk.awscore.internal.AwsExecutionContextBuilder.runInitialInterceptors(AwsExecutionContextBuilder.java:319)
	at software.amazon.awssdk.awscore.internal.AwsExecutionContextBuilder.invokeInterceptorsAndCreateExecutionContext(AwsExecutionContextBuilder.java:155)
	at software.amazon.awssdk.awscore.client.handler.AwsSyncClientHandler.invokeInterceptorsAndCreateExecutionContext(AwsSyncClientHandler.java:67)
	at software.amazon.awssdk.core.internal.handler.BaseSyncClientHandler.lambda$execute$1(BaseSyncClientHandler.java:76)
	at software.amazon.awssdk.core.internal.handler.BaseSyncClientHandler.measureApiCallSuccess(BaseSyncClientHandler.java:182)
	at software.amazon.awssdk.core.internal.handler.BaseSyncClientHandler.execute(BaseSyncClientHandler.java:74)
	at software.amazon.awssdk.core.client.handler.SdkSyncClientHandler.execute(SdkSyncClientHandler.java:45)
	at software.amazon.awssdk.awscore.client.handler.AwsSyncClientHandler.execute(AwsSyncClientHandler.java:53)
	at software.amazon.awssdk.services.s3.DefaultS3Client.listObjectsV2(DefaultS3Client.java:8724)
	at software.amazon.awssdk.services.s3.paginators.ListObjectsV2Iterable$ListObjectsV2ResponseFetcher.nextPage(ListObjectsV2Iterable.java:154)
	at software.amazon.awssdk.services.s3.paginators.ListObjectsV2Iterable$ListObjectsV2ResponseFetcher.nextPage(ListObjectsV2Iterable.java:145)
	at software.amazon.awssdk.core.pagination.sync.PaginatedResponsesIterator.next(PaginatedResponsesIterator.java:58)
	at software.amazon.awssdk.core.pagination.sync.PaginatedItemsIterable$ItemsIterator.<init>(PaginatedItemsIterable.java:58)
	at software.amazon.awssdk.core.pagination.sync.PaginatedItemsIterable.iterator(PaginatedItemsIterable.java:48)
	at java.base/java.lang.Iterable.spliterator(Iterable.java:101)
	at software.amazon.awssdk.core.pagination.sync.SdkIterable.stream(SdkIterable.java:34)
	at io.trino.filesystem.s3.S3FileSystem.listObjects(S3FileSystem.java:263)
	... 26 more

trino:product_s3> select * from remote_irc.product_s3.page_views_local;
         view_time          | user_id |          page_url          |     ds     | country 
----------------------------+---------+----------------------------+------------+---------
 2023-12-01 09:20:45.000000 |  123458 | http://example.com/contact | 2023-12-01 | Canada  
 2023-12-01 08:15:30.000000 |  123457 | http://example.com/about   | 2023-12-01 | USA     
 2023-12-01 10:25:50.000000 |  123459 | http://example.com/blog    | 2023-12-01 | UK      
(3 rows)

Query 20260429_065123_00016_brhru, FINISHED, 1 node
http://34.138.225.171:8080/ui/query.html?20260429_065123_00016_brhru
Splits: 3 total, 3 done (100.00%)
CPU Time: 0.6s total,     5 rows/s, 4.35KB/s, 32% active
Per Node: 0.2 parallelism,     1 rows/s,   891B/s
Parallelism: 0.2
Peak Memory: 1.18KB
2.86 [3 rows, 2.49KB] [1 rows/s, 891B/s]

```

### Error message and/or stacktrace

```
Query 20260429_064726_00014_brhru failed: Failed checking new table's location: s3://irc-e2e-env1-metalake-b-warehouse/env1/metalake_b/catalog_1/product_s3/page_views-c146b228bfd646889d20aeab7db054b0
io.trino.spi.TrinoException: Failed checking new table's location: s3://irc-e2e-env1-metalake-b-warehouse/env1/metalake_b/catalog_1/product_s3/page_views-c146b228bfd646889d20aeab7db054b0
	at io.trino.plugin.iceberg.IcebergMetadata.beginCreateTable(IcebergMetadata.java:1318)
	at io.trino.plugin.iceberg.IcebergMetadata.createTable(IcebergMetadata.java:1196)
	at io.trino.plugin.base.classloader.ClassLoaderSafeConnectorMetadata.createTable(ClassLoaderSafeConnectorMetadata.java:461)
	at io.trino.tracing.TracingConnectorMetadata.createTable(TracingConnectorMetadata.java:388)
	at io.trino.metadata.MetadataManager.createTable(MetadataManager.java:878)
	at io.trino.tracing.TracingMetadata.createTable(TracingMetadata.java:436)
	at io.trino.execution.CreateTableTask.internalExecute(CreateTableTask.java:306)
	at io.trino.execution.CreateTableTask.execute(CreateTableTask.java:128)
	at io.trino.execution.CreateTableTask.execute(CreateTableTask.java:94)
	at io.trino.execution.DataDefinitionExecution.start(DataDefinitionExecution.java:152)
	at io.trino.execution.SqlQueryManager.createQuery(SqlQueryManager.java:284)
	at io.trino.dispatcher.LocalDispatchQuery.startExecution(LocalDispatchQuery.java:150)
	at io.trino.dispatcher.LocalDispatchQuery.lambda$waitForMinimumWorkers$1(LocalDispatchQuery.java:134)
	at io.airlift.concurrent.MoreFutures.lambda$addSuccessCallback$0(MoreFutures.java:570)
	at io.airlift.concurrent.MoreFutures$3.onSuccess(MoreFutures.java:545)
	at com.google.common.util.concurrent.Futures$CallbackListener.run(Futures.java:1132)
	at io.trino.$gen.Trino_478____20260428_160509_2.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
	at java.base/java.lang.Thread.run(Thread.java:1474)
Caused by: io.trino.filesystem.TrinoFileSystemException: Failed to list location: s3://irc-e2e-env1-metalake-b-warehouse/env1/metalake_b/catalog_1/product_s3/page_views-c146b228bfd646889d20aeab7db054b0
	at io.trino.filesystem.s3.S3FileSystem.listObjects(S3FileSystem.java:270)
	at io.trino.filesystem.s3.S3FileSystem.listFiles(S3FileSystem.java:240)
	at io.trino.filesystem.switching.SwitchingFileSystem.listFiles(SwitchingFileSystem.java:116)
	at io.trino.filesystem.tracing.TracingFileSystem.lambda$listFiles$0(TracingFileSystem.java:119)
	at io.trino.filesystem.tracing.Tracing.withTracing(Tracing.java:51)
	at io.trino.filesystem.tracing.TracingFileSystem.listFiles(TracingFileSystem.java:119)
	at io.trino.filesystem.cache.CacheFileSystem.listFiles(CacheFileSystem.java:106)
	at io.trino.plugin.iceberg.IcebergMetadata.beginCreateTable(IcebergMetadata.java:1309)
	... 19 more
Caused by: software.amazon.awssdk.core.exception.SdkClientException: Unable to load credentials from any of the providers in the chain AwsCredentialsProviderChain(credentialsProviders=[SystemPropertyCredentialsProvider(), EnvironmentVariableCredentialsProvider(), WebIdentityTokenCredentialsProvider(), ProfileCredentialsProvider(profileName=default, profileFile=ProfileFile(sections=[])), ContainerCredentialsProvider(), InstanceProfileCredentialsProvider()]) : [SystemPropertyCredentialsProvider(): Unable to load credentials from system settings. Access key must be specified either via environment variable (AWS_ACCESS_KEY_ID) or system property (aws.accessKeyId)., EnvironmentVariableCredentialsProvider(): Unable to load credentials from system settings. Access key must be specified either via environment variable (AWS_ACCESS_KEY_ID) or system property (aws.accessKeyId)., WebIdentityTokenCredentialsProvider(): Either the environment variable AWS_WEB_IDENTITY_TOKEN_FILE or the javaproperty aws.webIdentityTokenFile must be set., ProfileCredentialsProvider(profileName=default, profileFile=ProfileFile(sections=[])): Profile file contained no credentials for profile 'default': ProfileFile(sections=[]), ContainerCredentialsProvider(): Cannot fetch credentials from container - neither AWS_CONTAINER_CREDENTIALS_FULL_URI or AWS_CONTAINER_CREDENTIALS_RELATIVE_URI environment variables are set., InstanceProfileCredentialsProvider(): Failed to load credentials from IMDS.]
	at software.amazon.awssdk.core.exception.SdkClientException$BuilderImpl.build(SdkClientException.java:130)
	at software.amazon.awssdk.auth.credentials.AwsCredentialsProviderChain.resolveCredentials(AwsCredentialsProviderChain.java:130)
	at software.amazon.awssdk.auth.credentials.internal.LazyAwsCredentialsProvider.resolveCredentials(LazyAwsCredentialsProvider.java:45)
	at software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider.resolveCredentials(DefaultCredentialsProvider.java:134)
	at software.amazon.awssdk.auth.credentials.AwsCredentialsProvider.resolveIdentity(AwsCredentialsProvider.java:54)
	at software.amazon.awssdk.services.s3.auth.scheme.internal.S3AuthSchemeInterceptor.lambda$trySelectAuthScheme$6(S3AuthSchemeInterceptor.java:169)
	at software.amazon.awssdk.core.internal.util.MetricUtils.reportDuration(MetricUtils.java:81)
	at software.amazon.awssdk.services.s3.auth.scheme.internal.S3AuthSchemeInterceptor.trySelectAuthScheme(S3AuthSchemeInterceptor.java:169)
	at software.amazon.awssdk.services.s3.auth.scheme.internal.S3AuthSchemeInterceptor.selectAuthScheme(S3AuthSchemeInterceptor.java:87)
	at software.amazon.awssdk.services.s3.auth.scheme.internal.S3AuthSchemeInterceptor.beforeExecution(S3AuthSchemeInterceptor.java:67)
	at software.amazon.awssdk.core.interceptor.ExecutionInterceptorChain.lambda$beforeExecution$1(ExecutionInterceptorChain.java:59)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1604)
	at software.amazon.awssdk.core.interceptor.ExecutionInterceptorChain.beforeExecution(ExecutionInterceptorChain.java:59)
	at software.amazon.awssdk.awscore.internal.AwsExecutionContextBuilder.runInitialInterceptors(AwsExecutionContextBuilder.java:319)
	at software.amazon.awssdk.awscore.internal.AwsExecutionContextBuilder.invokeInterceptorsAndCreateExecutionContext(AwsExecutionContextBuilder.java:155)
	at software.amazon.awssdk.awscore.client.handler.AwsSyncClientHandler.invokeInterceptorsAndCreateExecutionContext(AwsSyncClientHandler.java:67)
	at software.amazon.awssdk.core.internal.handler.BaseSyncClientHandler.lambda$execute$1(BaseSyncClientHandler.java:76)
	at software.amazon.awssdk.core.internal.handler.BaseSyncClientHandler.measureApiCallSuccess(BaseSyncClientHandler.java:182)
	at software.amazon.awssdk.core.internal.handler.BaseSyncClientHandler.execute(BaseSyncClientHandler.java:74)
	at software.amazon.awssdk.core.client.handler.SdkSyncClientHandler.execute(SdkSyncClientHandler.java:45)
	at software.amazon.awssdk.awscore.client.handler.AwsSyncClientHandler.execute(AwsSyncClientHandler.java:53)
	at software.amazon.awssdk.services.s3.DefaultS3Client.listObjectsV2(DefaultS3Client.java:8724)
	at software.amazon.awssdk.services.s3.paginators.ListObjectsV2Iterable$ListObjectsV2ResponseFetcher.nextPage(ListObjectsV2Iterable.java:154)
	at software.amazon.awssdk.services.s3.paginators.ListObjectsV2Iterable$ListObjectsV2ResponseFetcher.nextPage(ListObjectsV2Iterable.java:145)
	at software.amazon.awssdk.core.pagination.sync.PaginatedResponsesIterator.next(PaginatedResponsesIterator.java:58)
	at software.amazon.awssdk.core.pagination.sync.PaginatedItemsIterable$ItemsIterator.<init>(PaginatedItemsIterable.java:58)
	at software.amazon.awssdk.core.pagination.sync.PaginatedItemsIterable.iterator(PaginatedItemsIterable.java:48)
	at java.base/java.lang.Iterable.spliterator(Iterable.java:101)
	at software.amazon.awssdk.core.pagination.sync.SdkIterable.stream(SdkIterable.java:34)
	at io.trino.filesystem.s3.S3FileSystem.listObjects(S3FileSystem.java:263)
	... 26 more

```

### How to reproduce

1. Create table for remote IRC (S3 object store)

### Additional context

_No response_

--- a/iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/iceberg/service/CatalogWrapperForREST.java
+++ b/iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/iceberg/service/CatalogWrapperForREST.java
@@ -686,32 +686,32 @@ public class CatalogWrapperForREST extends IcebergCatalogWrapper {
     properties.putAll(request.properties());
 
     Map<String, String> config = Maps.newHashMap();
-    String location;
+    Catalog.TableBuilder tableBuilder =
+        loadedCatalog
+            .buildTable(ident, request.schema())
+            .withPartitionSpec(request.spec())
+            .withSortOrder(request.writeOrder())
+            .withProperties(properties);
+
+    Table table;
     if (request.location() != null) {
-      location = request.location();
+      table = tableBuilder.withLocation(request.location()).createTransaction().table();
     } else {
-      Table table =
-          loadedCatalog
-              .buildTable(ident, request.schema())
-              .withPartitionSpec(request.spec())
-              .withSortOrder(request.writeOrder())
-              .withProperties(properties)
-              .createTransaction()
-              .table();
-      Map<String, String> tableProperties = retrieveFileIOProperties(table.io());
-      config.putAll(
-          MapUtils.getFilteredMap(
-              tableProperties, key -> catalogPropertiesToClientKeys.contains(key)));
-      config.putAll(CredentialPropertyUtils.filterCredentialProperties(tableProperties));
-      location = table.location();
+      table = tableBuilder.createTransaction().table();
     }
+
+    Map<String, String> tableProperties = retrieveFileIOProperties(table.io());
+    config.putAll(
+        MapUtils.getFilteredMap(
+            tableProperties, key -> catalogPropertiesToClientKeys.contains(key)));
+    config.putAll(CredentialPropertyUtils.filterCredentialProperties(tableProperties));
 
     TableMetadata metadata =
         TableMetadata.newTableMetadata(
             request.schema(),
             request.spec() != null ? request.spec() : PartitionSpec.unpartitioned(),
             request.writeOrder() != null ? request.writeOrder() : SortOrder.unsorted(),
-            location,
+            table.location(),
             properties);
 
     return LoadTableResponse.builder().withTableMetadata(metadata).addAllConfig(config).build();

--- a/iceberg/iceberg-rest-server/src/test/java/org/apache/gravitino/iceberg/service/TestCatalogWrapperForREST.java
+++ b/iceberg/iceberg-rest-server/src/test/java/org/apache/gravitino/iceberg/service/TestCatalogWrapperForREST.java
@@ -22,11 +22,11 @@ package org.apache.gravitino.iceberg.service;
 import static org.mockito.Mockito.any;
 import static org.mockito.Mockito.anyMap;
 import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
 import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
 
 import com.google.common.collect.ImmutableMap;
-import java.lang.reflect.Method;
 import java.util.Map;
 import java.util.concurrent.atomic.AtomicBoolean;
 import org.apache.gravitino.catalog.lakehouse.iceberg.IcebergConstants;
@@ -190,7 +190,7 @@ public class TestCatalogWrapperForREST {
 
   @Test
   void testStageTableCreateWithLocationIncludesFileIoProperties() throws Exception {
-    Catalog catalog = mock(Catalog.class);
+    RESTCatalog catalog = mock(RESTCatalog.class);
     Catalog.TableBuilder tableBuilder = mock(Catalog.TableBuilder.class);
     Transaction transaction = mock(Transaction.class);
     Table table = mock(Table.class);
@@ -230,18 +230,58 @@ public class TestCatalogWrapperForREST {
             .stageCreate()
             .build();
 
-    Method method =
-        CatalogWrapperForREST.class.getDeclaredMethod(
-            "stageTableCreateInternal", Namespace.class, CreateTableRequest.class);
-    method.setAccessible(true);
-    LoadTableResponse response =
-        (LoadTableResponse) method.invoke(wrapper, Namespace.of("db"), request);
+    LoadTableResponse response = wrapper.createTable(Namespace.of("db"), request, false);
 
     Assertions.assertEquals(
         "org.apache.iceberg.aws.s3.S3FileIO", response.config().get(IcebergConstants.IO_IMPL));
     Assertions.assertEquals(
         "http://localhost:9000", response.config().get(IcebergConstants.ICEBERG_S3_ENDPOINT));
     verify(tableBuilder).withLocation("s3://bucket/warehouse/table");
+  }
+
+  @Test
+  void testStageTableCreateWithNullLocationDoesNotCallWithLocation() {
+    RESTCatalog catalog = mock(RESTCatalog.class);
+    Catalog.TableBuilder tableBuilder = mock(Catalog.TableBuilder.class);
+    Transaction transaction = mock(Transaction.class);
+    Table table = mock(Table.class);
+    FileIO fileIO = mock(FileIO.class);
+    when(catalog.buildTable(any(TableIdentifier.class), any())).thenReturn(tableBuilder);
+    when(tableBuilder.withPartitionSpec(any())).thenReturn(tableBuilder);
+    when(tableBuilder.withSortOrder(any())).thenReturn(tableBuilder);
+    when(tableBuilder.withProperties(anyMap())).thenReturn(tableBuilder);
+    when(tableBuilder.createTransaction()).thenReturn(transaction);
+    when(transaction.table()).thenReturn(table);
+    when(table.io()).thenReturn(fileIO);
+    when(table.location()).thenReturn("s3://bucket/warehouse/default-location");
+    when(fileIO.properties())
+        .thenReturn(
+            ImmutableMap.of(
+                IcebergConstants.IO_IMPL,
+                "org.apache.iceberg.aws.s3.S3FileIO",
+                IcebergConstants.ICEBERG_S3_ENDPOINT,
+                "http://localhost:9000"));
+
+    IcebergConfig config =
+        new IcebergConfig(
+            ImmutableMap.of(
+                IcebergConstants.CATALOG_BACKEND,
+                "memory",
+                IcebergConstants.WAREHOUSE,
+                "/tmp/warehouse"));
+    CatalogWrapperForREST wrapper = new StaticCatalogWrapperForREST("test", config, catalog);
+
+    Schema schema = new Schema(Types.NestedField.required(1, "id", Types.IntegerType.get()));
+    CreateTableRequest request =
+        CreateTableRequest.builder().withName("tbl").withSchema(schema).stageCreate().build();
+
+    LoadTableResponse response = wrapper.createTable(Namespace.of("db"), request, false);
+
+    Assertions.assertEquals(
+        "org.apache.iceberg.aws.s3.S3FileIO", response.config().get(IcebergConstants.IO_IMPL));
+    Assertions.assertEquals(
+        "http://localhost:9000", response.config().get(IcebergConstants.ICEBERG_S3_ENDPOINT));
+    verify(tableBuilder, never()).withLocation(any());
   }
 
   private static class LazyCheckCatalogWrapperForREST extends CatalogWrapperForREST {

--- a/iceberg/iceberg-rest-server/src/test/java/org/apache/gravitino/iceberg/service/TestCatalogWrapperForREST.java
+++ b/iceberg/iceberg-rest-server/src/test/java/org/apache/gravitino/iceberg/service/TestCatalogWrapperForREST.java
@@ -19,16 +19,29 @@
 
 package org.apache.gravitino.iceberg.service;
 
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.anyMap;
 import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
 
 import com.google.common.collect.ImmutableMap;
+import java.lang.reflect.Method;
 import java.util.Map;
 import java.util.concurrent.atomic.AtomicBoolean;
 import org.apache.gravitino.catalog.lakehouse.iceberg.IcebergConstants;
 import org.apache.gravitino.iceberg.common.IcebergConfig;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.Transaction;
 import org.apache.iceberg.catalog.Catalog;
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.io.FileIO;
 import org.apache.iceberg.rest.RESTCatalog;
+import org.apache.iceberg.rest.requests.CreateTableRequest;
+import org.apache.iceberg.rest.responses.LoadTableResponse;
+import org.apache.iceberg.types.Types;
 import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
 
@@ -175,6 +188,62 @@ public class TestCatalogWrapperForREST {
     }
   }
 
+  @Test
+  void testStageTableCreateWithLocationIncludesFileIoProperties() throws Exception {
+    Catalog catalog = mock(Catalog.class);
+    Catalog.TableBuilder tableBuilder = mock(Catalog.TableBuilder.class);
+    Transaction transaction = mock(Transaction.class);
+    Table table = mock(Table.class);
+    FileIO fileIO = mock(FileIO.class);
+    when(catalog.buildTable(any(TableIdentifier.class), any())).thenReturn(tableBuilder);
+    when(tableBuilder.withPartitionSpec(any())).thenReturn(tableBuilder);
+    when(tableBuilder.withSortOrder(any())).thenReturn(tableBuilder);
+    when(tableBuilder.withProperties(anyMap())).thenReturn(tableBuilder);
+    when(tableBuilder.withLocation("s3://bucket/warehouse/table")).thenReturn(tableBuilder);
+    when(tableBuilder.createTransaction()).thenReturn(transaction);
+    when(transaction.table()).thenReturn(table);
+    when(table.io()).thenReturn(fileIO);
+    when(table.location()).thenReturn("s3://bucket/warehouse/table");
+    when(fileIO.properties())
+        .thenReturn(
+            ImmutableMap.of(
+                IcebergConstants.IO_IMPL,
+                "org.apache.iceberg.aws.s3.S3FileIO",
+                IcebergConstants.ICEBERG_S3_ENDPOINT,
+                "http://localhost:9000"));
+
+    IcebergConfig config =
+        new IcebergConfig(
+            ImmutableMap.of(
+                IcebergConstants.CATALOG_BACKEND,
+                "memory",
+                IcebergConstants.WAREHOUSE,
+                "/tmp/warehouse"));
+    CatalogWrapperForREST wrapper = new StaticCatalogWrapperForREST("test", config, catalog);
+
+    Schema schema = new Schema(Types.NestedField.required(1, "id", Types.IntegerType.get()));
+    CreateTableRequest request =
+        CreateTableRequest.builder()
+            .withName("tbl")
+            .withSchema(schema)
+            .withLocation("s3://bucket/warehouse/table")
+            .stageCreate()
+            .build();
+
+    Method method =
+        CatalogWrapperForREST.class.getDeclaredMethod(
+            "stageTableCreateInternal", Namespace.class, CreateTableRequest.class);
+    method.setAccessible(true);
+    LoadTableResponse response =
+        (LoadTableResponse) method.invoke(wrapper, Namespace.of("db"), request);
+
+    Assertions.assertEquals(
+        "org.apache.iceberg.aws.s3.S3FileIO", response.config().get(IcebergConstants.IO_IMPL));
+    Assertions.assertEquals(
+        "http://localhost:9000", response.config().get(IcebergConstants.ICEBERG_S3_ENDPOINT));
+    verify(tableBuilder).withLocation("s3://bucket/warehouse/table");
+  }
+
   private static class LazyCheckCatalogWrapperForREST extends CatalogWrapperForREST {
 
     LazyCheckCatalogWrapperForREST(String catalogName, IcebergConfig config) {
@@ -187,6 +256,20 @@ public class TestCatalogWrapperForREST {
         throw new AssertionError("Catalog should not be loaded during wrapper construction");
       }
       return super.getCatalog();
+    }
+  }
+
+  private static class StaticCatalogWrapperForREST extends CatalogWrapperForREST {
+    private final Catalog catalog;
+
+    StaticCatalogWrapperForREST(String catalogName, IcebergConfig config, Catalog catalog) {
+      super(catalogName, config);
+      this.catalog = catalog;
+    }
+
+    @Override
+    public Catalog getCatalog() {
+      return catalog;
     }
   }
 }

